### PR TITLE
Documentation for travis-ci framework uploads

### PR DESCRIPTION
**Pull request by flovilmart**
- Small addition to the README in order to document how to setup travis releases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,6 +167,55 @@ carthage archive YourFrameworkName
...
+1. [Setup](https://docs.travis-ci.com/user/getting-started/) travis-ci for your repository (Steps 1 and 2)
+1. Create `.travis.yml` file at the root of your repository based on that template. Set `XCODE_SCHEME` as well as `FRAMEWORK_NAME` to your correct values
+
+	```
+	language: objective-c
+	osx_image: xcode7.1
+	env:
+		global: 
+		- XCODE_SCHEME=
+		- FRAMEWORK_NAME=
```
Comment (robrix) [inline README.md]: Maybe we should put placeholders here, and call the placeholder values about in the above list instead.

Comment (flovilmart) [inline README.md]: What do you mean? something like:

```
XCODE_SCHEME=MyScheme
FRAMEWORK_NAME=MyFramework
```

And say Replace MyScheme by the name of your Scheme and MyFramework by the name of your Framework?

Comment (robrix) [inline README.md]: Yes, but I was thinking a much more obvious placeholder, in the vein of `<THIS_IS_A_PLACEHOLDER_REPLACE_ME>` ^_^

Comment (robrix) [inline README.md]: That said the README currently uses `YourFrameworkName` in the archive section, so we can probably go with that.

**Comment by NachoSoto** [comment]
This is great!

File: README.md (hunk 1 above)
Comment (flovilmart) [inline README.md]: I can still update with our suggestion

**Comment by flovilmart** [comment]
@NachoSoto next step is `carthage setup travis`

File: README.md [hunk 2]
```diff
@@ -167,6 +167,55 @@ carthage archive YourFrameworkName
...
+	env:
+		global: 
+		- XCODE_SCHEME=<THIS_IS_A_PLACEHOLDER_REPLACE_ME>
+		- FRAMEWORK_NAME=<THIS_IS_A_PLACEHOLDER_REPLACE_ME>
+	before_install:
+	- brew install carthage
+	before_script:
+	# update the dependencies for the project
+	# you can remove if you don't have dependencies
+	- carthage update
```
Comment (neilpa) [inline README.md]: This should probably default to `carthage bootstrap`

Comment (flovilmart) [inline README.md]: I never really got the difference between `bootstrap` and `update`

Comment (mdiep) [inline README.md]: It should definitely be `bootstrap`.

`update` updates `Cartfile.resolved` to the newest version of all dependencies and then `bootstrap`s.

Comment (neilpa) [inline README.md]: `bootstrap` syncs exactly what's in the `Cartfile.resolved`. `update` could pick a newer dependency if you've specified something like `>= 1.0` and there's a new version available.

File: README.md [hunk 3]
```diff
@@ -167,6 +167,55 @@ carthage archive YourFrameworkName
...
+		- XCODE_SCHEME=<THIS_IS_A_PLACEHOLDER_REPLACE_ME>
+		- FRAMEWORK_NAME=<THIS_IS_A_PLACEHOLDER_REPLACE_ME>
+	before_install:
+	- brew install carthage
+	before_script:
+	# bootstrap the dependencies for the project
+	# you can remove if you don't have dependencies
+	- carthage bootstrap	
+	script:
+	- true
```
Comment (mdiep) [inline README.md]: Why is this needed? Can we remove this line and include `script:` in the commented-out lines below?

Comment (flovilmart) [inline README.md]: I'm not sure travis will like without the `script` instruction.

I can try

File: README.md [hunk 4]
```diff
@@ -167,6 +167,55 @@ carthage archive YourFrameworkName
...
+	# uncomment those line to run tests:
+	# - xcodebuild -scheme $XCODE_SCHEME -destination 'platform=iOS Simulator,name=iPhone 6,OS=9.1' -sdk iphonesimulator test
```
Comment (mdiep) [inline README.md]: Actually, these lines shouldn't be needed at all, right? If the `script:` item is removed above, then Travis should automatically run the tests.

File: README.md [hunk 5]
```diff
@@ -167,6 +167,55 @@ carthage archive YourFrameworkName
...
+1. [Install travis CLI](https://github.com/travis-ci/travis.rb#installation) with `gem install travis`
+1. [Setup](https://docs.travis-ci.com/user/getting-started/) travis-ci for your repository (Steps 1 and 2)
+1. Create `.travis.yml` file at the root of your repository based on that template. Set `XCODE_SCHEME` as well as `FRAMEWORK_NAME` to your correct values
+
+	```
+	language: objective-c
+	osx_image: xcode7.1
+	env:
+		global: 
+		- XCODE_SCHEME=<THIS_IS_A_PLACEHOLDER_REPLACE_ME>
```
Comment (mdiep) [inline README.md]: This should use the builtin `xcode_scheme` value. And I think we should demonstrate using `xcode_workspace` too.

File: README.md (hunk 4 above)
Comment (flovilmart) [inline README.md]: So that forces the user to have tests and use the default configuration.
I am not sure about all that I'm checking now.

File: README.md (hunk 5 above)
Comment (flovilmart) [inline README.md]: Actually we don't need it for building the framework. If we want to have multiple options, let's put some gists there for the non travis aficionados.

Comment (mdiep) [inline README.md]: Setting `xcode_scheme` and `xcode_workspace` should let us get rid of `script:` altogether IIUC.

Comment (flovilmart) [inline README.md]: As per: 
https://docs.travis-ci.com/user/languages/objective-c#Default-Test-Script

I'll update

**Comment by mdiep** [comment]
:fireworks:

**Comment by flovilmart** [comment]
Can't wait to have those in release :) 4 frameworks pending :)

**Comment by flovilmart** [comment]
Actually, we should have also: 

`xcode_sdk: iphonesimulator9.1`

 for example as otherwise the build fails

**Comment by mdiep** [comment]
Hmm… I didn't think that was necessary. But if it is, a PR would be great.

**Comment by flovilmart** [comment]
Yes it is :) just had a bad surprise trying it out :)
